crates/driver/src/infra/api/routes/settle/mod.rs

```diff
@@ -14,13 +14,15 @@ pub(in crate::infra::api) fn settle(router: axum::Router<State>) -> axum::Router
 
 async fn route(
     state: axum::extract::State<State>,
-    _: axum::Json<dto::Solution>,
+    solution: axum::Json<dto::Solution>,
 ) -> Result<(), (hyper::StatusCode, axum::Json<Error>)> {
     let competition = state.competition();
     let auction_id = competition.auction_id().map(|id| id.0);
     let handle_request = async {
         observe::settling();
-        let result = competition.settle().await;
+        let result = competition
+            .settle(solution.submission_deadline_latest_block)
```

Comment:
Seems a bit odd to have the submission deadline on the solution instead of the competition. 🤔 
I guess the reason is that the competition gets created before the `/settle` call happens which is where we learn about the deadline?
Ideally this would get changed because this conceptually allows the existence of 2 solutions for the same auction with different deadlines.

Reply:
@MartinquaXD the deadline is on `/settle` payload meaning that the autopilot is telling the driver how much time it has to settle it. I believe this was the main point from Haris. What is your suggestion here? :thinking:

Reply:
Misread the code because the `solution` variable name was a bit misleading. This actually the `/settle` requst. So the value is where it should be the naming of the variable and struct is just a bit confusing.